### PR TITLE
CFY-5821 add snapshots with pg

### PR DESCRIPTION
**Pull request by dusking**

### CODE REVIEW

File: workflows/cloudify_system_workflows/postgres.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
+import os
```
Comment (dankilman) [inline workflows/cloudify_system_workflows/postgres.py]: add license

File: workflows/cloudify_system_workflows/snapshot.py [hunk 2]
```diff
@@ -99,62 +102,166 @@ def chunks(g):
...
-def _copy_data(archive_root, config, to_archive=True):
-    """
-    Copy files/dirs between snapshot/manager and manager/snapshot.
+@workflow(system_wide=True)
+def create(snapshot_id, config, **kwargs):
+    ctx.logger.info('Creating snapshot..')
+    update_status = get_rest_client().snapshots.update_status
+    config = DictToAttributes(config)
+    try:
+        _create_snapshot(snapshot_id, config, **kwargs)
```
Comment (dankilman) [inline workflows/cloudify_system_workflows/snapshot.py]: run_create_snapshot

File: workflows/cloudify_system_workflows/snapshot.py [hunk 3]
```diff
@@ -99,62 +102,166 @@ def chunks(g):
...
-                    if os.path.isdir(s):
-                        shutil.copytree(s, d)
-                    else:
-                        shutil.copy2(s, d)
+
+@workflow(system_wide=True)
+def restore(snapshot_id, recreate_deployments_envs, config, force, timeout,
+            **kwargs):
+    ctx.logger.info('Restoring snapshot {0}'.format(snapshot_id))
+
```
Comment (dankilman) [inline workflows/cloudify_system_workflows/snapshot.py]: no enters

File: workflows/cloudify_system_workflows/snapshot.py [hunk 4]
```diff
@@ -99,62 +102,166 @@ def chunks(g):
...
+        if from_version.greater_than(manager_version):
+            raise NonRecoverableError(
+                'Cannot restore a newer manager\'s snapshot on this manager '
+                '[{0} > {1}]'.format(str(from_version), str(manager_version)))
+
+        existing_deployments_ids = [d.id for d in client.deployments.list()]
+        ctx.logger.info('Starting restoring snapshot of manager {0}'
+                        .format(from_version))
+
+        snapshot_from_manager_4 = from_version.equals(manager_version)
```
Comment (dankilman) [inline workflows/cloudify_system_workflows/snapshot.py]: renames

File: workflows/cloudify_system_workflows/snapshot.py [hunk 5]
```diff
@@ -712,28 +639,35 @@ def _restore_agents_data(tempdir):
...
 
-    new_plugins = _restore_elasticsearch(tempdir, es, metadata,
-                                         elasticsearch_read_timeout)
+    _restore_elasticsearch(tempdir, es, metadata,
+                           elasticsearch_read_timeout)
+
+    plugins = []
+
+    # postgres
+    if is_postgres:
```
Comment (dankilman) [inline workflows/cloudify_system_workflows/snapshot.py]: if from version

File: workflows/cloudify_system_workflows/utils.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
+import os
```
Comment (dankilman) [inline workflows/cloudify_system_workflows/utils.py]: license

File: workflows/cloudify_system_workflows/utils.py [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+import shlex
+import shutil
+import subprocess
+from cloudify.workflows import ctx
```
Comment (dankilman) [inline workflows/cloudify_system_workflows/utils.py]: enter

**Comment by dusking** [comment]
Reviewed by Dan

**Pull request merged**